src/D2L.Security.OAuth2/Keys/JsonWebKey.cs

```diff
@@ -49,91 +51,169 @@ public abstract class JsonWebKey {
 		/// <returns>A JWK DTO</returns>
 		public abstract object ToJwkDto();
 
+
 		/// <summary>
 		/// Deserialize a JWK
 		/// </summary>
 		/// <param name="json">The json JWK</param>
 		/// <returns>A <see cref="JsonWebKey"/></returns>
 		public static JsonWebKey FromJson( string json ) {
+			if( !TryParseJsonSigningKey( json, out var result, out var error, out var e, out _ ) ) {
+				throw new JsonWebKeyParseException( error, e );
+			}
+
+			return result;
+		}
+
+		public static bool TryParseJsonSigningKey(
+			string json,
+			[NotNullWhen( true )]
+			out JsonWebKey? key,
+			[NotNullWhen( false )]
+			out string? error,
+			[NullWhen( true )]
+			out Exception? exception,
+			out bool useEncKey
+		) {

```

Comment:
Mostly changed to TryX from throwing, but also added this separate `useEncKey` signal.